control-plane-operator/controllers/hostedcontrolplane/registryoperator/reconcile.go

```diff
@@ -0,0 +1,467 @@
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/metrics",
+							Port:   intstr.FromInt(metricsPort),
+							Scheme: corev1.URISchemeHTTPS,
+						},
+					},
+					InitialDelaySeconds: 15,
+					PeriodSeconds:       60,
+					SuccessThreshold:    1,
+					FailureThreshold:    3,
+					TimeoutSeconds:      5,
+				},
+			},
+			LivenessProbes: config.LivenessProbes{
+				containerMain().Name: {
+					ProbeHandler: corev1.ProbeHandler{
+						HTTPGet: &corev1.HTTPGetAction{
+							Path:   "/metrics",
+							Port:   intstr.FromInt(metricsPort),
+							Scheme: corev1.URISchemeHTTPS,
+						},
+					},
+					InitialDelaySeconds: 60,
+					PeriodSeconds:       60,
+					SuccessThreshold:    1,
+					FailureThreshold:    5,
+					TimeoutSeconds:      5,
+				},
+			},
+			Resources: config.ResourcesSpec{
+				containerMain().Name: {
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("50Mi"),
+						corev1.ResourceCPU:    resource.MustParse("10m"),
+					},
+				},
+				containerClientTokenMinter().Name: {
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10m"),
+						corev1.ResourceMemory: resource.MustParse("10Mi"),
+					},
+				},
+				containerWebIdentityTokenMinter().Name: {
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("10m"),
+						corev1.ResourceMemory: resource.MustParse("10Mi"),
+					},
+				},
+			},
+		},
+	}
+	params.deploymentConfig.SetDefaults(hcp, selectorLabels(), pointer.Int(1))
+	params.deploymentConfig.SetReleaseImageAnnotation(hcp.Spec.ReleaseImage)
+	return params
+}
+
+func ReconcileDeployment(deployment *appsv1.Deployment, params Params) error {
+	startScript, err := operatorStartScript()
+	if err != nil {
+		return fmt.Errorf("cannot create start script: %w", err)
+	}
+	deployment.Spec = appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: selectorLabels(),
+		},
+		Strategy: appsv1.DeploymentStrategy{
+			Type: appsv1.RecreateDeploymentStrategyType,
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: selectorLabels(),
+			},
+			Spec: corev1.PodSpec{
+				AutomountServiceAccountToken: pointer.Bool(false),
+				Containers: []corev1.Container{
+					util.BuildContainer(containerMain(), buildMainContainer(params.operatorImage, params.registryImage, params.prunerImage, params.releaseVersion, startScript)),
+					util.BuildContainer(containerClientTokenMinter(), buildClientTokenMinter(params.tokenMinterImage, params.issuerURL)),
+				},
+				Volumes: []corev1.Volume{
+					util.BuildVolume(volumeClientToken(), buildVolumeClientToken),
+					util.BuildVolume(volumeServingCert(), buildVolumeServingCert),
+					util.BuildVolume(volumeAdminKubeconfig(), buildVolumeAdminKubeconfig),
+					util.BuildVolume(volumeCABundle(), buildVolumeCABundle),
+				},
+			},
+		},
+	}
+
+	switch params.platform {
+	case hyperv1.AWSPlatform:
```

Comment:
so on azure this isn't going to work initially since there are no creds? If so, lets maybe create a small reminder jira issue to fix that up later?

Reply:
The creds for azure are already on the guest cluster since the hcco puts them there. The operator should read them as usual. We have a special case here for AWS because we need a token minter instead of using the token request volume in the podspec